src/frontend/src/expr/utils.rs

```diff
@@ -523,6 +523,31 @@ impl WatermarkAnalyzer {
                 }
                 _ => WatermarkDerivation::None,
             },
+            ExprType::AddWithTimeZone | ExprType::SubtractWithTimeZone => {
+                // Requires time zone and interval to be literal, at least for now.
+                let time_zone = match &func_call.inputs()[2] {
+                    ExprImpl::Literal(lit) => lit.get_data().as_ref().map(|s| s.as_utf8()),
+                    _ => return WatermarkDerivation::None,
+                };
+                let interval = match &func_call.inputs()[1] {
+                    ExprImpl::Literal(lit) => lit.get_data().as_ref().map(|s| s.as_interval()),
+                    _ => return WatermarkDerivation::None,
+                };
+                // null zone or null interval is treated same as const `interval '1' second`, to be
+                // consistent with other match arms.
+                let zone_without_dst = time_zone.map_or(true, |s| s.eq_ignore_ascii_case("UTC"));
```

Comment:
seems very useful and can be more accurate :thinking: maybe we should maintain it as a specific function?

Reply:
I agree, but still not sure about the proper signature: does it take a `&str`, `ScalarImpl`, `Datum`, `Literal` or even `ExprImpl`? Is it always okay to treat null as no-dst?